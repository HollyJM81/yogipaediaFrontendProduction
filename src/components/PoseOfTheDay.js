import React, { useState, useEffect } from "react";
import axios from "axios";
import "../styles/pose-of-the-day.scss";

const PoseOfTheDay = () => {
  const [randomImage, setRandomImage] = useState(null);

  useEffect(() => {
    axios
      .get("https://yoga-api-nzy4.onrender.com/v1/poses")
      .then((response) => {
        const yogaPosesArray = response.data;
        const randomIndex = Math.floor(Math.random() * yogaPosesArray.length); //randomIndex is the index of a random object from yogaPosesArray
        setRandomImage(yogaPosesArray[randomIndex].url_png); //sets randomImage to the url of the random pose object
      });
  }, []);

  return (
    <div>
      <h2>Pose Of The Day</h2>
      <div>
        <img src={randomImage} alt="yoga pose of the day" />
      </div>
      <div>
        <p>
          Yoga is a holistic practice that originated in ancient India and
          encompases physical postures, breathing exercises, meditation, and
          ethical principles. It is a discipline aimed at promoting physical,
          mental, and spiritual well-being. Through the practice of yoga,
          individuals can improve flexibility, strength, and balance, while also
          reducing stress, enhancing mental clarity and fostering a sense of
          inner peace. Whether you are a beginner yogi or an aspiring yoga
          teacher, you can explore a variety of yoga poses by using the
          Yogipedia app.
        </p>
      </div>
    </div>
  );
};

export default PoseOfTheDay;
