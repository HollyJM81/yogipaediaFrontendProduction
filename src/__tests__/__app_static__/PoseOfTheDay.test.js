import PoseOfTheDay from " ../../components/app_static/PoseOfTheDay";
