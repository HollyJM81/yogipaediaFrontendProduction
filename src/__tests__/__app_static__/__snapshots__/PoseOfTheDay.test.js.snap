// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoseOfTheDay renders correctly 1`] = `
<DocumentFragment>
  <div
    class="page-content__container"
  >
    <div
      class="pose__container"
    >
      <div
        class="pose-of-the-day"
      >
        <h2>
          Pose Of The Day
        </h2>
      </div>
      <div
        class="pose__image"
      >
        <img
          alt="yoga pose of the day"
        />
      </div>
    </div>
    <div
      class="pose-description__container"
    >
      <p>
        Yoga is a holistic practice that originated in ancient India and encompases physical postures, breathing exercises, meditation, and ethical principles. It is a discipline aimed at promoting physical, mental, and spiritual well-being. Through the practice of yoga, individuals can improve flexibility, strength, and balance, while also reducing stress, enhancing mental clarity and fostering a sense of inner peace. Whether you are a beginner yogi or an aspiring yoga teacher, you can explore a variety of yoga poses by using the Yogipedia app.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
